bundles/org.openhab.binding.globalcache/src/main/java/org/openhab/binding/globalcache/internal/handler/GlobalCacheHandler.java

```diff
@@ -554,14 +547,16 @@ private void writeCommandToDevice(RequestMessage requestMessage) throws IOExcept
          */
         private String readReplyFromDevice(RequestMessage requestMessage) throws IOException {
             logger.trace("Processor for thing {} reading reply from device", thingID());
-
             if (connectionManager.getCommandIn() == null) {
                 logger.debug("Error reading from device because input stream object is null");
                 return "ERROR: BufferedReader is null!";
```

Comment:
I see now that this kind of error handling was also used here, so consider if this should also be turned into something like:
`throw new IOException("BufferedReader is null");`

Reply:
See my reply above. I was hoping to fix a clear error condition rather than rework a bunch of code. Not to say that it shouldn't be reworked at some point. 😉

Reply:
Yeah, I posted this comment seconds before your reply to the other comment, resolving this one. :)

Reply:
Thanks. Yeah, at some point I definitely need to come back around and spend some time on this binding. I'm pretty sure I can do what needs to be done with a lot less code. 😉

Reply:
And I also was feeling a little time pressure with the release just around the corner.